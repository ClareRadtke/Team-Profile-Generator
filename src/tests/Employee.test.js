const { Employee } = require("../lib/classes");

describe(Employee, () => {
  it("should create a new Employee instance", () => {
    const employee = new Employee("John Doe", "123", "foo@example.com");
    expect(employee).toMatchSnapshot();
  });
  describe("getName()", () => {
    it("should return the name", () => {
      const employee = new Employee("John Doe", "123", "foo@example.com");
      expect(employee.getName()).toBe("John Doe");
    });
  });
  describe("getId()", () => {
    it("should return the id", () => {
      const employee = new Employee("John Doe", "123", "foo@example.com");
      expect(employee.getId()).toBe("123");
    });
  });
  describe("getEmail()", () => {
    it("should return the email", () => {
      const employee = new Employee("John Doe", "123", "foo@example.com");
      expect(employee.getEmail()).toBe("foo@example.com");
    });
  });
  describe("getRole()", () => {
    it("should return the role", () => {
      const employee = new Employee("John Doe", "123", "foo@example.com");
      expect(employee.getRole()).toBe("Employee");
    });
  });
});
