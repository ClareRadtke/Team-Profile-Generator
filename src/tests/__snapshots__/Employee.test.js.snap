// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Employee should create a new Employee instance 1`] = `
Employee {
  "email": "foo@example.com",
  "id": "123",
  "name": "John Doe",
}
`;
